### DISCUSSION TITLE
Does rtabmap have semantic mapping?

### DISCUSSION THREAD

**Discussion by Rodhaaret** [introlab/rtabmap#1254] [Ideas]
Hey everyone!

I have used rtabmap in the past for different SLAM approaches on different mobile platform. Now im working on a new project in which we are looking into semantic mapping in general. I'm curious to know if rtabmap already implements different semantic mapping approaches?

Such as attaching detected objects to the map or defining regions based on seen object. Something like Kimera: 
https://github.com/MIT-SPARK/Kimera-Semantics

Hope my question makes sense!

**Comment by matlabbe** [comment]
Short answer: it doesn't.

Long answer: we can feed `user_data` to rtabmap with any information that need to be attached to the graph. The `user_data` will only make sense to external code defining them, and won't be used internally in rtabmap for loop closure detection or re-localization. rtabmap will publish the `user_data` with the graph, thus external code would need to read and merge duplicates if there are (e..g, frame A and frame B saw a table at a position relative to them at a some time, it could be the same table if the resulting pose in map frame are close together). The `user_data` could be objects detected with some machine learning approach.

**Reply by Rodhaaret** [reply]
Thanks a lot for your reply! We will have a look at this!
